File: 00_intro/I6_perlin_noise/typescript/src/sketch.ts
import p5 from "p5";

let t = 0;

new p5((p: p5) => {

  p.setup = () => {
    p.createCanvas(640,360);

    // Defaults
    p.stroke(255);
    p.strokeWeight(2);
    p.noFill();
  };

  p.draw = () => {
    // Update
    t += 0.02; // adjust speed (the lower, the slower)

    p.background(0) // clear previous render
    let xOff = t;
    p.beginShape()
    for (let x = 0; x < p.width; x++) {
      const y = p.noise(xOff) * p.height;
      p.vertex(x, y);
      xOff += 0.005; // adjust flattening (the lower, the flatter)
    }
    p.endShape()
  };
});

File: 00_intro/I6_perlin_noise/java/src/main/java/org/example/Sketch.java
package org.example;

import processing.core.PApplet;

public class Sketch extends PApplet {
    public static final double X_SPEED = 0.01; // this affects the x scroll speed: the lower, the slower.
    public static final double FLATTENING = 0.002; // this value affects the flattening: the lower, the flatter
    private float t = 0;

    @Override
    public void settings() {
        size(640, 360);
    }

    @Override
    public void setup() {
        windowTitle("Perlin Noise");

        stroke(255);
        strokeWeight(2);
        noFill();
    }

    @Override
    public void draw() {
        update();
        background(0); // clear previous render
        drawGraph();
    }

    private void update() {
        t += X_SPEED;
    }

    private void drawGraph() {
        float xoff = t;
        beginShape();
        for (int x = 0; x < width; x++) {
            float y = noise(xoff) * height;
            vertex(x, y);
            xoff += FLATTENING;
        }
        endShape();
    }
}
